### PR TITLE
None type object cannot call close

### PR DESCRIPTION
**Pull request by sek788432**
if the log_file_name argument is wrong, self.file is None.
And it will raise none type object cannot call close()

### CODE REVIEW

File: flaml/training_log.py [hunk 1]
```diff
@@ -159,6 +161,8 @@ def training_log_writer(filename: str):
         w = TrainingLogWriter(filename)
         w.open()
         yield w
+    except FileNotFoundError:
```
Comment (ekzhu) [inline flaml/training_log.py:164]: We can remove this except block as the error message below doesn't provide more detailed information than the builtin error message.

File: flaml/training_log.py [hunk 2]
```diff
@@ -169,5 +173,7 @@ def training_log_reader(filename: str):
         r = TrainingLogReader(filename)
         r.open()
         yield r
+    except FileNotFoundError:
```
Comment (ekzhu) [inline flaml/training_log.py:176]: Same here.

**Review by ekzhu — COMMENTED** [review]
Thanks for the contribution! The error messages for file not exists is not much different from the builtin message. Perhaps we can remove the except blocks.

File: flaml/training_log.py (hunk 1 above)
Comment (sek788432) [reply flaml/training_log.py:164]: I got it. It seems reasonable. Do I need to update another commit to modify it?

Comment (ekzhu) [reply flaml/training_log.py:164]: Yes. Please update with another commit. Thanks!

Comment (sek788432) [reply flaml/training_log.py]: I updated. Thanks!

**Review by ekzhu — APPROVED** [review]
Thanks!

**Pull request merged**